test/curve_analysis/test_curve_fit.py

```diff
@@ -63,6 +64,67 @@ class TestAnalysis(CurveAnalysis):
     return TestAnalysis()
 
 
+class TestFitData(QiskitTestCase):
+    """Unittest for fit data dataclass."""
+
+    def test_get_value(self):
+        """Get fit value from fit data object."""
+        data = FitData(
+            popt=np.asarray([1.0, 2.0, 3.0]),
+            popt_keys=["a", "b", "c"],
+            popt_err=np.asarray([0.1, 0.2, 0.3]),
+            pcov=np.diag(np.ones(3)),
+            reduced_chisq=0.0,
+            dof=0,
+            x_range=(0, 0),
+            y_range=(0, 0),
+        )
+
+        a_val = data.fitval("a")
+        self.assertEqual(a_val, FitVal(1.0, 0.1))
+
+        a_val = data.fitval("b")
+        self.assertEqual(a_val, FitVal(2.0, 0.2))
+
+        a_val = data.fitval("c")
+        self.assertEqual(a_val, FitVal(3.0, 0.3))
+
+    def test_get_value_with_unit_scaling(self):
+        """Get fit value from fit data object with unit and aux unit."""
+
+        data = FitData(
+            popt=np.logspace(-7, 7, 15),
+            popt_keys=[f"x{i}" for i in range(15)],
+            popt_err=np.full(15, None),
+            pcov=np.diag(np.ones(15)),
+            reduced_chisq=0.0,
+            dof=0,
+            x_range=(0, 0),
+            y_range=(0, 0),
+        )
+
+        unitrefs = [
+            FitVal(100.0, unit="nHz"),
+            FitVal(1.0, unit="μHz"),
+            FitVal(10.0, unit="μHz"),
+            FitVal(100.0, unit="μHz"),
+            FitVal(1.0, unit="mHz"),
+            FitVal(10.0, unit="mHz"),
+            FitVal(100.0, unit="mHz"),
+            FitVal(1.0, unit="Hz"),
+            FitVal(10.0, unit="Hz"),
+            FitVal(100.0, unit="Hz"),
+            FitVal(1.0, unit="kHz"),
+            FitVal(10.0, unit="kHz"),
+            FitVal(100.0, unit="kHz"),
+            FitVal(1.0, unit="MHz"),
+            FitVal(10.0, unit="MHz"),
+        ]
+
+        for i in range(15):
```

Comment:
15 is hard-coded, replace with `len(unitrefs)` or `len(popt)` (maybe verify that they're equal?)

Reply:
This test is removed since scaling is not performed with `FitData`